UPGRADING.md

```diff
@@ -5,6 +5,12 @@ Note, always read the **SimApp** section for more information on application wir
 
 ## [Unreleased]
 
+### Migration to Collections
```

Comment:
This will be removed in the backport.

Reply:
Hmm id like to add some use cases migrating to collections. Maybe a new PR though?

Reply:
nice, ill add them separately then since this is tied to a specific issue.